romancal/patch_match/patch_match.py

```diff
@@ -0,0 +1,299 @@
+    dec = ptab[:]["dec_center"]
+    # # Convert all celestial coordinates to cartesion coordinates.
+    vec_centers = np.array(sgv.lonlat_to_vector(ra, dec)).transpose()
+    # # Organize corners into two ra, dec lists
+    # imra = np.array([point[0] for point in image_corners])
+    # imdec = np.array([point[1] for point in image_corners])
+    vec_im_corners = image_coords_to_vec(image_corners)
+    # Approximate center of image by averaging corner vectors
+    im_center = normalize_vector(vec_im_corners.mean(axis=1))
+    # Compute difference vector between image center and patch centers
+    diff = vec_centers - im_center
+    dist = np.sqrt((diff**2).sum(axis=1)) * 180 / np.pi
+    match = np.where(dist < 0.5)
+    ncandidates = len(match[0])
+    # Now see which of these that are close actually overlap the supplied image.
+    # (Is it necessary to check that the corners are in a sensible order?)
+    # All the corner coordinates are returned as arrays.
+    mra1 = ptab[match]["ra_corn1"]
+    mra2 = ptab[match]["ra_corn2"]
+    mra3 = ptab[match]["ra_corn3"]
+    mra4 = ptab[match]["ra_corn4"]
+    mdec1 = ptab[match]["dec_corn1"]
+    mdec2 = ptab[match]["dec_corn2"]
+    mdec3 = ptab[match]["dec_corn3"]
+    mdec4 = ptab[match]["dec_corn4"]
+    mcenters = vec_centers[match]
+    mra = np.vstack([mra1, mra2, mra3, mra4, mra1])
+    mdec = np.vstack([mdec1, mdec2, mdec3, mdec4, mdec1])
+    points = np.array(sgv.lonlat_to_vector(mra, mdec))
+    # Create polygon for supplied image_corners
+    pvec_im_corners = np.zeros((5, 3))
+    pvec_im_corners[:4] = vec_im_corners.transpose()
+    pvec_im_corners[4] = pvec_im_corners[0]
+    impoly = sgp.SingleSphericalPolygon(pvec_im_corners, im_center)
+    realmatch = []
+    for i in range(ncandidates):
+        # print(i)
+        cellpoints = points[:, :, i].transpose()
+        cellcenter = mcenters[i]
+        cellpoly = sgp.SingleSphericalPolygon(cellpoints, cellcenter)
+        if impoly.intersects_poly(cellpoly):
+            # print(f"candidate {i} intersects")
+            realmatch.append(i)
+    return match[0][realmatch], match[0]
+
+
+def get_cartesian_corners(patch):
+    """
+    Construct a the vertex coordinates for a patch from a patch definition suitable
+    for plotting the defined region (ra coordinates, dec coordinates). It returns
+    coordinates in the cartesian system so that it avoids the distortions in plots
+    due to areas approaching the poles.
+    """
+    corners = (
+        (
+            patch["dec_corn1"],
+            patch["dec_corn2"],
+            patch["dec_corn3"],
+            patch["dec_corn4"],
+            patch["dec_corn1"],
+        ),
+        (
+            patch["ra_corn1"],
+            patch["ra_corn2"],
+            patch["ra_corn3"],
+            patch["ra_corn4"],
+            patch["ra_corn1"],
+        ),
+    )
+    corners = np.array(corners)
+    vec_corners = sgv.lonlat_to_vector(corners[0], corners[1])
```

Comment:
Even if you do not like my suggestion with list comprehension and `zip`, my not create two separate lists `ra_corners` and `dec_corners` instead of punching them into a single array and then extracting rows?